File: Arrays/Easy/35/35.c
int searchInsert(int* nums, int numsSize, int target){
    int low = 0, high = numsSize - 1;
    int mid;
	if(target<nums[0]){
		return 0;
	}
	if(target>nums[high]){
		return high + 1;
	}
    while(low <= high){
        mid = (low + high) / 2;
        if(nums[mid] == target){
            return mid;
        }else if(nums[mid] < target){
            low = mid + 1;
        }else{
            high = mid - 1;
        }
    }
    return low;
}

File: Arrays/Easy/35/35.py
from typing import List

class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        min = 0
        max = len(nums) - 1
        while min <= max:
            mid = (min + max) // 2
            if nums[mid] == target:
                return mid
            elif nums[mid] < target:
                min = mid + 1
            else:
                max = mid - 1
        return min